system/security/shared/caching.cpp

```diff
@@ -672,16 +680,55 @@ bool CPermissionsCache::queryPermsManagedFileScope(ISecUser& sec_user, const cha
 
 SecAccessFlags CPermissionsCache::queryDefaultPermission(ISecUser& user)
 {
-    if (m_defaultPermission == SecAccess_Unknown)
+
+    if (m_useLegacyDefaultFileScopePermissionCache)
     {
-        if (m_secMgr)
-            m_defaultPermission = m_secMgr->queryDefaultPermission(user);
+        if (m_defaultPermission == SecAccess_Unknown)
+        {
+            if (m_secMgr)
+                m_defaultPermission = m_secMgr->queryDefaultPermission(user);
+            else
+                m_defaultPermission = SecAccess_Full;
+
+            DBGLOG("Legacy default file scope permission set to %s(%d) for all users, based on User '%s'", getSecAccessFlagName(m_defaultPermission),
```

Comment:
This looks like this tracing could be excessive.

Reply:
That line will appear once whenever the cache is reset and when using the legacy code that has the error. I think it is important to leave in if the legacy option is chosen, then you can search back in the logs, if needed, to see what default scope permission is in effect.